### PR TITLE
Cst-5855 custom resource policies are duplicated if multiple upload configuration share the same access conditions

### LINKED ISSUE #8282: Custom resource policies are duplicated if multiple upload configuration share the same access conditions
**Describe the bug**
If you configure multiple upload step to use different metadata and/or access conditions when the user select an access condition that is shared among multiple configuration the corresponding resource policies is applied n times (1 for each configuration).
Due to the same bug it is possible to set an access condition in the submission also if not allowed by the specific submission configuration associated with the collection if this access condition is defined in another configuration

**To Reproduce**
1. in the `access-conditions.xml` duplicate the `uploadConfigurationDefault` bean configuration giving it a second name. Map also this bean in your `uploadConfigurationService`
2. make a submission and assign any access condition to a bitstream in your item
3. check the resource policy of the bitstream once the item has been archived, you will see two copies of the TYPE_CUSTOM resource policy selected during the submission

**Expected behavior**
Only one resourcepolicy should be created. Only access condition allowed by the specific submission configuration should be processed

### PR DESCRIPTION
**Pull request by ballarinie**
* Fixes https://github.com/DSpace/DSpace/issues/8282

## Description
This PR fixes the bug involving the bitstream of a submitted item having duplicate access conditions when the selected ones are shared among multiple configurations.
Also, requests related to unconfigured and/or unavailable access conditions are now managed and rejected with 422 status code.
ITs have been implemented by previously changing the corresponding `access-conditions.xml` accordingly.
.
## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/factory/impl/BitstreamResourcePolicyUtils.java [hunk 1]
```diff
@@ -66,24 +65,31 @@ public static void findApplyResourcePolicy(Context context, Iterator<UploadConfi
...
     public static void findApplyResourcePolicy(Context context, UploadConfiguration uploadConfiguration,
             DSpaceObject obj, String name, String description,
                                                Date startDate, Date endDate)
             throws SQLException, AuthorizeException, ParseException {
+        boolean found = false;
         for (AccessConditionOption aco : uploadConfiguration.getOptions()) {
             if (aco.getName().equalsIgnoreCase(name)) {
                 aco.createResourcePolicy(context, obj, name, description, startDate, endDate);
-                return;
+                found = true;
```
Comment (corrad82-4s) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/factory/impl/BitstreamResourcePolicyUtils.java:86]: once found, we could break the for loop

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java [hunk 2]
```diff
@@ -611,7 +611,7 @@ public void deleteOneForbiddenTest() throws Exception {
 
         Item item = witem.getItem();
 
-        //Add a bitstream to the item
+        //Add
```
Comment (corrad82-4s) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java:614]: In my opinion, this comment is not 100% clear

**Review by corrad82-4s — CHANGES_REQUESTED** [review]
@ballarinie , code overall looks good to me. I dropped a couple of minor notes. 
If you can, please add a comment with a couple of instructions about how this PR can be tested, if a correspondent angular PR is needed, etc.

**Comment by ballarinie** [comment]
Fixed code according to @corrad82-4s 's comments.

About testing:
- to properly test the duplicated resource policy bug is solved, I simply replicated the steps Andrea Bollini https://github.com/DSpace/DSpace/issues/8282. Now, only one resource policy is going to be shown when checking those of the bitstream of the related archived item.
- this PR also denies patch operations that would add unconfigured and/or unavailable access conditions to the item during a submission. This could be tested via postman (quoting the workspaceitem-data-upload.md rest contract):
`curl --data '{[ { "op": "add", "path": "/sections/<:name-of-the-form>/files/<:file-idx>/accessConditions/-", "value": {name: "...", <wrong-access-condition>: ".."}}]}' -X PATCH ${dspace7-url}/api/submission/workspaceitems/<:id>`
Using an unconfigured/unexisting access-condition should bring to a 422 answer. This situation is not handled in the UI, but the list of the selectable access conditions is limited to those configured in the access-conditions.xml file.

**Review by corrad82-4s — APPROVED** [review]
@tdonohue Mi feedback has been addressed and instructions about how to test this PR have been added. I approve the work

**Review by tdonohue — APPROVED** [review]
👍 Thanks @ballarinie !  I tested this today by trying to reproduce the original bug, and I can no longer do so.  The code and tests also look good.

**Pull request merged**